silo-to-silo/contracts/src/SiloToSilo.sol

```diff
@@ -0,0 +1,280 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {IERC20 as IERC20_NEAR} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {AuroraSdk, NEAR, PromiseCreateArgs, PromiseResultStatus, PromiseWithCallback} from "@auroraisnear/aurora-sdk/aurora-sdk/AuroraSdk.sol";
+import "@auroraisnear/aurora-sdk/aurora-sdk/Utils.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/Base64.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import "./IEvmErc20.sol";
+
+struct TokenInfo {
+    string nearTokenAccountId;
+    bool isStorageRegistered;
+}
+
+contract SiloToSilo is Initializable, UUPSUpgradeable, AccessControlUpgradeable, PausableUpgradeable {
+    using AuroraSdk for NEAR;
+    using AuroraSdk for PromiseCreateArgs;
+    using AuroraSdk for PromiseWithCallback;
+
+    bytes32 public constant CALLBACK_ROLE = keccak256("CALLBACK_ROLE");
+
+    uint64 constant BASE_NEAR_GAS = 10_000_000_000_000;
+    uint64 constant WITHDRAW_NEAR_GAS = 50_000_000_000_000;
+    uint64 constant FT_TRANSFER_CALL_NEAR_GAS = 150_000_000_000_000;
+    uint128 constant ASCII_0 = 48;
+    uint128 constant ASCII_9 = 57;
+    uint128 constant ONE_YOCTO = 1;
+    uint128 constant NO_DEPOSIT = 0;
+
+    NEAR public near;
+    string public siloAccountId;
+
+    //[auroraErc20Token] => tokenAccountIdOnNear
+    mapping(IEvmErc20 => TokenInfo) public registeredTokens;
+
+    //[auroraErc20Token][userAddressOnAurora] => userBalance
+    mapping(IEvmErc20 => mapping(address => uint128)) public balance;
+
+    event TokenRegistered(IEvmErc20 token, string nearAccountId);
```

Comment:
Should we create more events? Also for storage deposit, withdraw and for ft_transfer_call?